src/honeybadger/context.cr

```diff
@@ -0,0 +1,34 @@
+module Honeybadger
+  def self.context(**args) : Nil
+    Context.current.store(args)
+  end
+
+  class Context
+    def initialize
+      @data = Hash(String, String).new
+    end
+
+    delegate to_s, to: @data
+    forward_missing_to @data
+
+    def self.current : self
+      Fiber.current.honeybadger_context
+    end
```

Comment:
This is really slick. One of the things I was most concerned about with the proposed API was that if you did a bunch of work inside child fibers (say, to fan out work to perform it concurrently), you wouldn’t get the parent fiber’s context if an exception were raised in one of the child fibers. For example:

```crystal
def call(http_context)
  channel = Channel(Result?).new

  widgets = WidgetsService.call(http_context.request.query_params[“search”])

  widgets.each do |widget|
    spawn do
      result = process(widget)
    rescue ex
      # FIXME: Missing parent fiber’s context!
      Honeybadger.notify ex
    ensure
      channel.send result
    end
  end

  widgets.each { channel.receive }
end
```

But I think your `Context.current` idea here makes this really simple to handle with 2 additional lines of code:

```diff
 def call(http_context)
   channel = Channel(Result?).new

   widgets = WidgetsService.call(http_context.request.query_params[“search”])

+  parent_context = Context.current
   widgets.each do |widget|
     spawn do
+      Context.current.merge! parent_context
       result = process(widget)
     rescue ex
       Honeybadger.notify ex
     ensure
       channel.send result
     end
   end

   widgets.each { channel.receive }
 end
```

Reply:
@jgaskins that's a good point which I hadn't considered. Perhaps that explicit inheritance is better anyway...or at least that's what we can tell them when they ask!